### ISSUE TITLE
Increasing the number of Rancher server pods causes the cluster agent to connection is closed

### ISSUE THREAD

**Issue by xiaoluhong** [rancher/rancher#17127]
**What kind of request is this (question/bug/enhancement/feature request):**
Question/Bug

**Environment information**
- Rancher version :
`rancher/rancher`:v2.x.x
- Installation option:
HA

**Steps to reproduce (least amount of steps as possible):**
1. Perform three-node HA installation according to   https://rancher.com/docs/rancher/v2.x/en/installation/ha/;
1. Resolve the domain name to one of the nodes;
1. Import an RKE K8S cluster；

    <img width="1083" alt="image" src="https://user-images.githubusercontent.com/20707603/50067356-b8c75100-01fb-11e9-9e13-b9c41da3a114.png">

1. Add session retention for rancher ingress：
    <img width="802" alt="image" src="https://user-images.githubusercontent.com/20707603/50066987-ae0bbc80-01f9-11e9-83ad-a43542e62298.png">

1. Scale rancher server pod to 4;

**Result:**
Increasing the number of Rancher server pods causes the cluster agent to connection is closed

<img width="1309" alt="image" src="https://user-images.githubusercontent.com/20707603/50067139-85d08d80-01fa-11e9-835c-ad0163f08bcb.png">

**Comment by xiaoluhong** [comment]
@orangedeng @pengjiang80 @loganhz

**Comment by superseb** [comment]
Please check the logging if this is the same as https://github.com/rancher/rancher/issues/16301

**Comment by alena1108** [comment]
@xiaoluhong does it close permanently, and never reconnect back? @superseb even if nginx drops the connection, rancher agent should handle it and restore the situation

**Comment by xiaoluhong** [comment]
@alena1108 Hi,It will automatically reconnect.   

I've also tested the external nginx L7 load balancer:
1. rancher was working in host network mode;
2.  the load balancer use IP hash mode (to make sure requests from the same client are sent to the same back-end server).  
3. I stopped one node firstly;
4. wait until cluster agent reconnect to rancher server successfully (on Rancher UI, the cluster status is `Active`);
5. restarted the closed node; 
6. however, the cluster status will be `Unavailable`
7. after about a few seconds (30s-60s), the cluster status will be `Active` again
